src/install.rs

```diff
@@ -113,23 +172,40 @@ fn write_disk(
     if ignition.is_some() || config.firstboot_kargs.is_some() || config.platform.is_some() {
         let mount = mount_boot(&config.device)?;
         if let Some(ignition) = ignition {
-            write_ignition(mount.mountpoint(), ignition)?;
+            write_ignition(mount.mountpoint(), &config.ignition_hash, ignition)
+                .chain_err(|| "writing Ignition configuration")?;
         }
         if let Some(firstboot_kargs) = config.firstboot_kargs.as_ref() {
-            write_firstboot_kargs(mount.mountpoint(), firstboot_kargs)?;
+            write_firstboot_kargs(mount.mountpoint(), firstboot_kargs)
+                .chain_err(|| "writing firstboot kargs")?;
         }
         if let Some(platform) = config.platform.as_ref() {
-            write_platform(mount.mountpoint(), platform)?;
+            write_platform(mount.mountpoint(), platform).chain_err(|| "writing platform ID")?;
         }
     }
 
     Ok(())
 }
 
 /// Write the Ignition config.
-fn write_ignition(mountpoint: &Path, mut config_in: File) -> Result<()> {
+fn write_ignition(
+    mountpoint: &Path,
+    digest_in: &Option<IgnitionHash>,
+    mut config_in: File,
+) -> Result<()> {
     eprintln!("Writing Ignition config");
 
+    // Verify configuration digest, if any.
+    if let Some(ref digest) = digest_in {
+        eprintln!("Writing Ignition config");
```

Comment:
We're now printing this twice.

Reply:
Ah good catch, this was actually a chunk of changes. I slightly moved the printing around.